### ISSUE TITLE
Non-posix ignore patterns are given to globby on win32 with new `nuxt generate` cache feature

### ISSUE THREAD

**Issue by lihbr** [nuxt/nuxt.js#7792]
### Versions

- nuxt: v2.14.0
- node: v12.13.1

### Reproduction

`N/A`

<details>
<summary>Example <code>./node_modules/.cache/nuxt/build.json</code> file generated on Windows</summary>
<br>

```json
{
  "nuxtVersion": "v2.14.0",
  "ssr": true,
  "target": "static",
  "snapshot": {
    "nuxt.config.js": {
      "checksum": "cd98971a"
    },
    "package.json": {
      "checksum": "a5162d54"
    },
    "components\\Logo.vue": {
      "checksum": "76f576d6"
    },
    "components\\README.md": {
      "checksum": "7d302e3d"
    },
    "assets\\README.md": {
      "checksum": "22131941"
    },
    "dist\\200.html": {
      "checksum": "e0ec4fad"
    },
    "dist\\favicon.ico": {
      "checksum": "425b0923"
    },
    "dist\\index.html": {
      "checksum": "f935f725"
    },
    "layouts\\default.vue": {
      "checksum": "f8cfc395"
    },
    "yarn.lock": {
      "checksum": "f57dd5de"
    },
    "layouts\\README.md": {
      "checksum": "f46f8d28"
    },
    "middleware\\README.md": {
      "checksum": "18f9be49"
    },
    "pages\\index.vue": {
      "checksum": "be074d11"
    },
    "pages\\README.md": {
      "checksum": "f2797eed"
    },
    "plugins\\README.md": {
      "checksum": "863ea810"
    },
    "store\\README.md": {
      "checksum": "25fae952"
    },
    "dist\\_nuxt\\runtime.249950c.js": {
      "checksum": "db130dae"
    },
    "dist\\_nuxt\\app.df91f1d.js": {
      "checksum": "847636ea"
    },
    "dist\\_nuxt\\pages\\index.d55436c.js": {
      "checksum": "df82ff7"
    },
    "dist\\_nuxt\\static\\1595883046\\payload.js": {
      "checksum": "b7213a5a"
    },
    "dist\\_nuxt\\commons\\app.356543e.js": {
      "checksum": "6a8077f2"
    },
    "dist\\README.md": {
      "checksum": "e5c7534"
    },
    "dist\\_nuxt\\vendors~app.b74a912.js": {
      "checksum": "c753c628"
    }
  }
}
```
</details>

<details>
<summary><code>nuxt generate</code> running webpack build because of change inside <code>./dist</code></summary>
<br>
<img src="https://user-images.githubusercontent.com/25330882/88593052-3fdadd00-d05f-11ea-994b-c64f61f0ad96.png" alt="nuxt generate running webpack build because of changes inside ./dist" />
</details>

### Steps to reproduce

On Windows:
1. `yarn create nuxt-app` or `npx create-nuxt-app` ;
2. just select `universal` & `static` ;
3. run `yarn generate` or `npm run generate` twice.

### What is Expected?

New generate cache feature should work on Windows.

### What is actually happening?

Snapshot contains files that should be ignored leading to comparison always detecting new changes from dist. Indeed globby and underlying fast-glob API both require patterns using forward-slashes only:
- https://github.com/sindresorhus/globby#api
- https://github.com/mrmlnc/fast-glob#pattern-syntax

The error is happening inside the [`snapshot` function](https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/utils/generate.js#L142) where globby receives its `ignore` option from [`nuxt.options.generate.cache.ignore`](https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/utils/generate.js#L51) which contains platform specific paths, example defaults on Windows:
```json
[
  "D:\\Libraries\\Desktop\\test\\node_modules\\.cache\\nuxt",
  "static",
  "D:\\Libraries\\Desktop\\test\\dist",
  "node_modules",
  ".**/*",
  ".*",
  "README.md"
]
```

### Fix suggestion

The most straightforward fix I can think of is to change [this line](https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/utils/generate.js#L51) with something like this:
```javascript
const snapshotOptions = {
  rootDir: nuxt.options.rootDir,
  ignore: nuxt.options.generate.cache.ignore.map(enforcePosixPath),
  globbyOptions: nuxt.options.generate.cache.globbyOptions
}
```

The tricky part comes to how to implement this `enforcePosixPath` function. Sadly node [`path`](https://nodejs.org/api/path.html) module doesn't provide a utility to easily convert any path (win32 or posix) to a posix one. I found [upath](https://www.npmjs.com/package/upath) which comes with a `normalize` method that should solve that, although it's probably not suitable to add a new dependency just for that and perhaps you already have something for that inside @nuxtjs/utils but I did not find such thing ^^

### Additional information

Hehe, it's me again, the Windows person! You might have seen me [fixing non-posix path before](https://github.com/nuxt-community/global-components/pull/15) and I'm here again to provide people running their CI on win32 platform the ability to enjoy the new cache feature just as well as posix peeps~ 😃

More seriously I don't know who's having their CI on win32 but for people working under Windows it's still nice to have cache for local "test" build~

Oh and I'd happily work on a PR if we agree on a fix implementation!

Cheers~

**Comment by pi0** [comment]
Hey @lihbr 😊 Thanks for detailed information. Of course this is a valid bug. PR more than welcome.

**Comment by lihbr** [comment]
Thanks @pi0, so going down the [upath](https://www.npmjs.com/package/upath) way? or do you have a better idea in mind? 🙂 I feel it'll raise a ton of edge cases if we (or at least I ^^') try to implement `win32ToPosix` ourselves 🤔

**Comment by pi0** [comment]
Should be fixed with [v2.14.1](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.1)
